File: experiment/src/test/scala/org/excavator/boot/experiment/test/InterproceduralAnalysis.scala
package org.excavator.boot.experiment.test

class InterproceduralAnalysis {

  private def absoluteDifference(x: Int, y:Int):Int = {
    if(y > x) y - x else x - y
  }

  def analyze():Unit = {
    val x = absoluteDifference(12, 15) + absoluteDifference(9, 2)
    if (x == 10){
      println(x)
    }
  }

}

File: experiment/src/test/java/org/excavator/boot/experiment/InterproceduralAnalysis.java
package org.excavator.boot.experiment;

public class InterproceduralAnalysis {
    private int absoluteDifference(int x, int y){
        if(y > x){
            return y - x;
        }else{
            return x - y;
        }
    }
    public void analyze(){
        int x= absoluteDifference(12, 15) + absoluteDifference(9, 2);
        if(x == 10){
            System.out.println(x);
        }
    }
}
